contracts/defi/provider/MakerProvider.sol

```diff
@@ -0,0 +1,818 @@
+     * @param _collateralAmount The amount of collateral to remove.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element.
+     */
+    function removeCollateral(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _collateral, 
+        uint256 _collateralAmount, 
+        address[] calldata _oracles
+    ) 
+        external 
+    {
+        require(_collateral == ETH_TOKEN_ADDRESS, "Maker: collateral must be ETH");
+        removeCollateral(_wallet, _loanId, _collateralAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Increases the debt by borrowing more token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     * @param _debtToken The token borrowed (must be the address of the DAI contract).
+     * @param _debtAmount The amount of token to borrow.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element.
+     */
+    function addDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        IMakerCdp makerCdp = IMakerCdp(_oracles[0]);
+        require(_debtToken == makerCdp.sai(), "Maker: debt token must be DAI");
+        addDebt(_wallet, _loanId, _debtAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Decreases the debt by repaying some token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     * @param _debtToken The token to repay (must be the address of the DAI contract).
+     * @param _debtAmount The amount of token to repay.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element and Uniswap's Factory as second.
+     */
+    function removeDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        IMakerCdp makerCdp = IMakerCdp(_oracles[0]);
+        require(_debtToken == makerCdp.sai(), "Maker: debt token must be DAI");
+        removeDebt(_wallet, _loanId, _debtAmount, makerCdp, UniswapFactory(_oracles[1]));
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the target CDP.
+     * @param _oracles An array containing the address of MakerDAO's Tub contract as first element.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated, 3: loan exists but we are unable to provide info] and the estimated ETH value of the loan
+     * combining all collaterals and all debts. When status = 1 it represents the value that could still be borrowed, while with status = 2
```

Comment:
"and the estimated ETH value of the loan combining all collaterals and all debts" -> "and a value (in ETH) representing the value that could still be borrowed when status = 1; or the value of the collateral that should be added to avoid liquidation when status = 2"

Reply:
Done.